### PR TITLE
Add asset channel as way to export assets

### PR DESCRIPTION
**Pull request by doluk**
By passing a channel to the export function, the assets get exported to this channel. Meaning deleting the channel of the transcript does not lead to missing assets in the transcripts/dead links.

### CODE REVIEW

File: chat_exporter/construct/message.py [hunk 1]
```diff
@@ -434,6 +457,7 @@ async def gather_messages(
     guild: discord.Guild,
     pytz_timezone,
     military_time,
+    channel: discord.TextChannel
```
Comment (mahtoid) [inline chat_exporter/construct/message.py:460]: Naming

File: chat_exporter/construct/message.py [hunk 2]
```diff
@@ -460,7 +484,8 @@ async def gather_messages(
             guild,
             meta_data,
             message_dict,
-        ).construct_message()
+            ).construct_message(channel)
```
Comment (mahtoid) [inline chat_exporter/construct/message.py:487]: You can pass this in to the class - much cleaner for access.

File: chat_exporter/construct/message.py [hunk 3]
```diff
@@ -241,11 +244,31 @@ async def build_sticker(self):
...
+                            data.seek(0)
+                            attach = discord.File(data, a.filename)
+                            msg: discord.Message = await channel.send(file=attach)
+                            a = msg.attachments[0]
+                except discord.errors.Forbidden as e:
+                    # permission error
+                    raise e
+                except discord.errors.HTTPException as e:
+                    # general http error
+                    raise e
```
Comment (mahtoid) [inline chat_exporter/construct/message.py:268]:
```
except (discord.errors.Forbidden, discord.errors.HTTPException) as e:
    raise e
```

File: chat_exporter/construct/message.py [hunk 4]
```diff
@@ -241,11 +244,31 @@ async def build_sticker(self):
...
+                except Exception as e:
+                    # deal with all other errors
+                    pass
```
Comment (mahtoid) [inline chat_exporter/construct/message.py:271]: Not fond of greedy `Exception`

**Comment by mahtoid** [comment]
Would you like to update the README as well to include this new optional argument?

File: chat_exporter/construct/message.py (hunk 3 above)
Comment (doluk) [reply chat_exporter/construct/message.py]: discord.errors.Forbidden is a subclass from HTTPException. I just added it explicitly to talk about if we want to deal with them differently, but imo we should raise them all to the user, so we could even remove the try ... except completely....

**Comment by doluk** [comment]
> Would you like to update the README as well to include this new optional argument?

Added it

File: chat_exporter/construct/message.py [hunk 5]
```diff
@@ -460,7 +479,8 @@ async def gather_messages(
             guild,
             meta_data,
             message_dict,
-        ).construct_message()
+            ).construct_message(asset_channel)
```
Comment (mahtoid) [inline chat_exporter/construct/message.py:482]: Pass it in to `MessageConstruct` (as part of the class) than the `construct_message` function

File: chat_exporter/construct/message.py (hunk 3 above)
Comment (mahtoid) [reply chat_exporter/construct/message.py]: If you are just doing `discord.errors.HTTPException` which raises anyways, you can remove the `try:except`. 
Alternatively, you could customise the error return. 

I'm still yet to test this locally.

File: chat_exporter/construct/message.py (hunk 5 above)
Comment (doluk) [reply chat_exporter/construct/message.py:482]: oh I forgot to remove that

**Pull request merged**